### PR TITLE
PLAT-1773: When clear button is clicked on zen-input a change event is triggered

### PR DESCRIPTION
**Pull request by f-jauregui**
[JIRA ticket](https://reciprocitylabs.atlassian.net/browse/)

⚠️ **NOTE: Nor onClick and onMouseup worked on this component, not sure if it's a Stencil bug.** ⚠️ 

#### Description
As per JIRA ticket

#### ChangeLOG

[Changelog](https://zen-ui.zengrc.com/?path=/docs/changelog--page)

#### Useful links
[Zen UI readme](https://github.com/reciprocity/zen-ui/blob/main/README.md)

[PR review guidelines](https://github.com/reciprocity/zengrc/blob/develop/doc/dev_environment_and_process/pull_request_reviews.md)

[Git commit guidelines](https://github.com/reciprocity/zengrc/blob/develop/doc/dev_environment_and_process/commit_guidelines.md)

### CODE REVIEW

**Review by chavlji — CHANGES_REQUESTED** [review]
Nice. Two comments:
1. `input` event should be fired as well. As by definition
2. _"The input event fires when the value of an <input>, <select>, or <textarea> element has been changed."_ - from [mdn](https://developer.mozilla.org/en-US/docs/Web/API/HTMLElement/input_event))
2. On date picker we get 2 change events on clear click. Could you please investigate why?

**Review by mihaelpalfi — APPROVED** [review]
Looks good. Change event triggered. Can confirm two events fired in date picker. I assume the issue is that it is logged for the input component and the date picker component. So story book.

**Comment by f-jauregui** [comment]
@chavlji I asked for your review again because as @mihaelpalfi said, I think the double change event is caused by Storybook.

**Review by chavlji — APPROVED** [review]
Ok. If you have some time, please investigate why it's called twice in the storybook.

**Comment by homer0** [comment]
It shouldn't have an `input` event; `input` is for user interaction, changing the value programmatically doesn't trigger it. This button is similar to `<input type="reset" />`, and if you listen for `input` and `change` events on an input that can be reset, the only triggered event is `change`.

Regarding the double event, is it possible that they are the `change` from the input and the `change` triggered from picker itself?

**Comment by f-jauregui** [comment]
> It shouldn't have an `input` event; `input` is for user interaction, changing the value programmatically doesn't trigger it. This button is similar to `<input type="reset" />`, and if you listen for `input` and `change` events on an input that can be reset, the only triggered event is `change`.
> 
> Regarding the double event, is it possible that they are the `change` from the input and the `change` triggered from picker itself?

@chavlji @homer0 Should I roll back the last commit?

File: src/components/zen-date-picker/tests/zen-date-picker.spec.tsx [hunk 1]
```diff
@@ -192,11 +181,6 @@ describe('zen-date-picker', () => {
     expect(JSON.stringify(datepicker.value)).toEqual('null');
   });
 
-  it('should not render x button if allowEmpty is false', async () => {
```
Comment (chavlji) [inline src/components/zen-date-picker/tests/zen-date-picker.spec.tsx:195]: I'd still test this? Test just if zen-input has attr `clear-button` set. Jic someone messes up your attr binding...

**Review by chavlji — APPROVED** [review]
Nice! Works as expected.
I'd just put back this test and then you can merge it! 🚢

File: src/components/zen-date-picker/tests/zen-date-picker.spec.tsx (hunk 1 above)
Comment (homer0) [reply src/components/zen-date-picker/tests/zen-date-picker.spec.tsx:195]: Agree, even if the button is on another component, we should test that the attr set on the picker reflects correctly on the input attrs.

**Review by homer0 — APPROVED** [review]

**Pull request merged**